x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -309,37 +467,81 @@ const installTransformsAssets = async (
     await Promise.all(
       transforms.map(async (transform) => {
         const index = transform.content.dest.index;
-        const pipelineId = transform.content.dest.pipeline;
 
+        const aliases = transformsSpecifications
+          .get(transform.transformModuleId)
+          ?.get('destinationIndexAlias');
         try {
-          await retryTransientEsErrors(
+          const resp = await retryTransientEsErrors(
             () =>
               esClient.indices.create(
                 {
                   index,
-                  ...(pipelineId ? { settings: { default_pipeline: pipelineId } } : {}),
+                  aliases,
                 },
                 { ignore: [400] }
               ),
             { logger }
           );
+          logger.debug(`Created destination index: ${index}`);
+
+          // If index already exists, we still need to update the destination index alias
+          // to point '{destinationIndexName}.latest' to the versioned index
+          // @ts-ignore status is a valid field of resp
+          if (resp.status === 400 && aliases) {
+            await retryTransientEsErrors(
+              () =>
+                esClient.indices.updateAliases({
+                  body: {
+                    actions: Object.keys(aliases).map((alias) => ({ add: { index, alias } })),
+                  },
+                }),
+              { logger }
+            );
+            logger.debug(`Created aliases for destination index: ${index}`);
+          }
         } catch (err) {
+          logger.debug(
```

Comment:
Should this be logged to `error` or is this a non-blocking issue?

Reply:
I have this same question about making errors visible to the user. It's generally a lot easier to debug with error messages than without!

Reply:
Updated here [`e1a0050` (#142920)](https://github.com/elastic/kibana/pull/142920/commits/e1a0050a8184b520a96ff9930d7c09b2636b8fa8)